app/models/post.rb

```diff
@@ -0,0 +1,13 @@
+class Post < ApplicationRecord
+  belongs_to :author, class_name: 'User'
+  has_many :likes
+  has_many :comments
+
+  def update_counter
+    author.update(post_counter: author.posts.count)
+  end
```

Comment:
## Optional:
- Please you can use one of the [Rails Active Callbacks](https://guides.rubyonrails.org/active_record_callbacks.html) - `after_save` to call the `#update_post_counter` method every time a new post has been saved to database, then make the `update_post_counter` method private. Example:
```ruby
 after_save :update_post_counter

  private
```

```ruby
def update_post_counter
    author.increment!(:posts_counter)
  end
```